packages/grafana-runtime/src/services/pluginExtensions/extensions.ts

```diff
@@ -1,34 +1,37 @@
-import { getPluginsExtensionRegistry, PluginsExtension } from './registry';
+import { type PluginExtension } from '@grafana/data';
 
-export type GetPluginExtensionsOptions = {
+import { getPluginsExtensionRegistry } from './registry';
+
+export type PluginExtensionsOptions<T extends object> = {
   placement: string;
+  context?: T;
 };
 
 export type PluginExtensionsResult = {
-  extensions: PluginsExtension[];
-  error?: Error;
+  extensions: PluginExtension[];
 };
 
-export class PluginExtensionsMissingError extends Error {
-  readonly placement: string;
-
-  constructor(placement: string) {
-    super(`Could not find extensions for '${placement}'`);
-    this.placement = placement;
-    this.name = PluginExtensionsMissingError.name;
-  }
-}
-
-export function getPluginExtensions({ placement }: GetPluginExtensionsOptions): PluginExtensionsResult {
+export function getPluginExtensions<T extends object = {}>(
+  options: PluginExtensionsOptions<T>
+): PluginExtensionsResult {
+  const { placement, context } = options;
   const registry = getPluginsExtensionRegistry();
-  const extensions = registry[placement];
+  const items = registry[placement] ?? [];
+
+  const extensions = items.reduce<PluginExtension[]>((result, item) => {
+    if (!context || !item.configure) {
+      result.push(item.extension);
+      return result;
+    }
 
-  if (!Array.isArray(extensions)) {
-    return {
-      extensions: [],
-      error: new PluginExtensionsMissingError(placement),
-    };
-  }
+    const extension = item.configure(context);
+    if (extension) {
+      result.push(extension);
+    }
+    return result;
+  }, []);
 
-  return { extensions };
+  return {
+    extensions: extensions,
+  };
```

Comment:
**Nit:** are we planning to return multiple different things here other than the list of extensions? (If not then maybe it would make sense to just return with the extensions object?)

Reply:
We might add more things :)